Framework/DataHandling/src/LoadNXcanSAS.cpp

```diff
@@ -568,5 +549,53 @@ void LoadNXcanSAS::exec() {
   setProperty("OutputWorkspace", ws);
 }
 
+void LoadNXcanSAS::loadTransmission(H5::Group &entry, const std::string &name) {
+  if (!hasTransmissionEntry(entry, name)) {
+    g_log.information("NXcanSAS file does not contain transmission for " +
+                      name);
+    return;
+  }
+
+  auto transmission =
+      entry.openGroup(sasTransmissionSpectrumGroupName + "_" + name);
+
+  // Create a 1D workspace
+  auto tDataSet = transmission.openDataSet(sasTransmissionSpectrumT);
+  auto workspace = createWorkspaceForHistogram(tDataSet);
+
+  // Load logs
+  loadLogs(entry, workspace);
+  auto title = workspace->getTitle();
+  const std::string transExtension = "_trans_" + name;
+  title += transExtension;
+  workspace->setTitle(transExtension);
+
+  // Load instrument
+  loadInstrument(entry, workspace);
+
+  // Load transmission data
+  loadTransmissionData(transmission, workspace);
+
+  // Set to distribution
+  workspace->setDistribution(true);
+  workspace->setYUnitLabel("Transmission");
+
+  // Set the workspace on the output
+
+  std::string propertyName;
+  if (name == "sample")
+    propertyName = "TransmissionWorkspace";
+  else
+    propertyName = "TransmissionCanWorkspace";
+  const std::string doc = "The transmission workspace";
+
+  declareProperty(
```

Comment:
I see - this is the best way to deal with that then. My main concern was the use of the ADS in the algorithm. I'm trying to limit this from occurring when I see it.

I will just give it a quick functional test with these changes and merge it if it's good then.